core/network/src/protocol/sync.rs

```diff
@@ -443,49 +452,41 @@ impl<B: BlockT> ChainSync<B> {
 
 	/// Request syncing for the given block from given set of peers.
 	// This is similar to on_block_announce with unknown parent hash.
-	pub fn sync_fork(&mut self, peers: Vec<PeerId>, hash: &B::Hash, number: NumberFor<B>) ->
-		Vec<(PeerId, BlockRequest<B>)>
-	{
+	pub fn sync_fork(&mut self, peers: Vec<PeerId>, hash: &B::Hash, number: NumberFor<B>) {
 		debug!(target: "sync", "Explicit sync request for block {:?} with {:?}", hash, peers);
-		if self.is_known(&hash) || self.is_already_downloading(&hash) {
+		if self.is_known(&hash) {
 			debug!(target: "sync", "Refusing to sync known hash {:?}", hash);
-			return Vec::default();
+			return;
 		}
 
 		let block_status = self.client.block_status(&BlockId::Number(number - One::one()))
 			.unwrap_or(BlockStatus::Unknown);
 		if block_status == BlockStatus::InChainPruned {
 			trace!(target: "sync", "Refusing to sync ancient block {:?}", hash);
-			return Vec::default();
+			return;
```

Comment:
this is also not the only kind of ancient fork we can encounter

Reply:
Sync attempts to download the chain up to the last finalized block, not just one block. There's currently a limit of how many blocks these forks thare downloaded up the chain can be. We can't be allowed to download infinite chains of headers/blocks without validating them.

Reply:
> Sync attempts to download the chain up to the last finalized block, not just one block

I don't understand what you mean. Yes, there has to be a practical cutoff somewhere, and I think refusing to download forks beyond `finalized_height + constant_allowance` would be reasonable (with `constant_allowance` around 1000). 

But the most recent finalized block of the fork is going to be the common ancestor (otherwise it wouldn't be a fork). So we always have to download beyond that.